src/shared/components/mutationTable/MutationTable.tsx

```diff
@@ -1049,4 +1084,30 @@ export default class MutationTable<
             />
         );
     }
+
+    protected getMutations() {
+        let data: Mutation[][] | undefined = [];
+        if (this.props.dataStore) {
+            data = this.props.dataStore.allData;
+        } else if (this.props.data) {
+            data = this.props.data;
+        }
+        return data;
+    }
+
+    protected hasRequiredASCNProperty(property: string): boolean {
+        let data = this.getMutations();
+        if (data) {
+            return data.some((row: Mutation[]) => {
+                return row.some((m: Mutation) => {
+                    return (
+                        m.alleleSpecificCopyNumber !== undefined &&
+                        (m.alleleSpecificCopyNumber as any)[property] !==
```

Comment:
@alisman @onursumer I never done any generic property check in this way. Is this valid?

Reply:
Following up on this - @zhx828 What is the recommended way to check for a generic property?

Reply:
@zhx828 I am adding documentation into the function explaining what this does. The data array is an array of table rows, and each row is an array of mutation records for that row (like when a patient has multiple samples containing the same mutation).
The logic is roughly:
"see if any row contains any mutation where the alleleSpecificCopyNumber subobject is present and contains a value for the key 'property'"
If this fails it means there is **no ASCN data with that property present at all**.

This is similar to the test here:
 https://github.com/cBioPortal/cbioportal-frontend/blob/fb7029b3be8758f80255d15875c0df0cee72fa30/src/pages/resultsView/ResultsViewPageStoreUtils.ts#L657

line 657 is testing whether there is some geneticProfile ... where every gene (line 660) was profiled for the sample (or not)